iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeManager.java

```diff
@@ -961,99 +950,80 @@ private void close(String processorName) throws FileNodeManagerException {
-          try {
-            close(processorEntry.getKey());
-          } catch (FileNodeManagerException e) {
-            throw e;
-          }
-        }
-      } catch (FileNodeManagerException e) {
-        throw new FileNodeManagerException(e);
-      } finally {
-        LOGGER.info("Close all filenode processor successfully");
-        fileNodeManagerStatus = FileNodeManagerStatus.NONE;
-      }
-    } else {
+    if (fileNodeManagerStatus != FileNodeManagerStatus.NONE) {
       LOGGER.info("Failed to close all filenode processor because of merge operation");
+      return;
+    }
+    fileNodeManagerStatus = FileNodeManagerStatus.CLOSE;
+    try {
+      for (Map.Entry<String, FileNodeProcessor> processorEntry : processorMap.entrySet()) {
+        close(processorEntry.getKey());
+      }
+    } finally {
+      LOGGER.info("Close all FileNodeProcessors ends");
+      fileNodeManagerStatus = FileNodeManagerStatus.NONE;
     }
   }
 
   /**
    * force flush to control memory usage.
    */
   public void forceFlush(BasicMemController.UsageLevel level) {
-    // TODO : for each FileNodeProcessor, call its forceFlush()
     // you may add some delicate process like below
     // or you could provide multiple methods for different urgency
     switch (level) {
+      // only select the most urgent (most active or biggest in size)
+      // processors to flush
+      // only select top 10% active memory user to flush
       case WARNING:
-        // only select the most urgent (most active or biggest in size)
-        // processors to flush
-        // only select top 10% active memory user to flush
         try {
           flushTop(0.1f);
         } catch (IOException e) {
-          LOGGER.error("force flush memory data error: {}", e.getMessage());
-          e.printStackTrace();
+          LOGGER.error("force flush memory data error: {}", e);
         }
         break;
+      // force all processors to flush
       case DANGEROUS:
-        // force all processors to flush
         try {
           flushAll();
         } catch (IOException e) {
-          LOGGER.error("force flush memory data error: {}", e.getMessage());
-          e.printStackTrace();
+          LOGGER.error("force flush memory data error: {}", e);
         }
         break;
+      // if the flush thread pool is not full ( or half full), start a new
+      // flush task
       case SAFE:
-        // if the flush thread pool is not full ( or half full), start a new
-        // flush task
         if (FlushManager.getInstance().getActiveCnt() < 0.5 * FlushManager.getInstance()
-            .getThreadCnt()) {
+                .getThreadCnt()) {
           try {
             flushTop(0.01f);
           } catch (IOException e) {
-            LOGGER.error("force flush memory data error:{}", e.getMessage());
-            e.printStackTrace();
+            LOGGER.error("force flush memory data error:{}", e);
```

Comment:
{} is unneccessary. There are many similar situations below.

Reply:
fixed